Kudu.Client/Infrastructure/HttpResponseResult.cs

```diff
@@ -8,11 +9,16 @@ public class HttpResponseResult<T>
     {
         public IDictionary<string, IEnumerable<string>> Headers { get; set; }
         public T Body { get; set; }
+        public HttpStatusCode Status { get; set; }
 
-        private HttpResponseResult(IDictionary<string, IEnumerable<string>> headers, T body)
+        private HttpResponseResult(
+            IDictionary<string, IEnumerable<string>> headers,
+            T body,
+            HttpStatusCode statusCode)
```

Comment:
yes.. only two difference, we have strongly Type and header is a dictionary.

but i think i can have extension methods on HttpResponseMessage to do the strong typing. let me update to HttpResponseMessage to see how it goes